File: Data Structures & Algorithms/find-pivot-index/submission-1.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        self.prefixSum = [0] * (n+1)
        for i in range(n):
            self.prefixSum[i+1] = self.prefixSum[i] + nums[i]
        print(self.prefixSum)

        for i in range(n):
            if ((self.prefixSum[n]-self.prefixSum[i+1]) == self.prefixSum[i]):
                return i
        return -1

File: Data Structures & Algorithms/permutations-ii/submission-1.cpp
class Solution {
public:
    vector<vector<int>> result;

    void backtrack(vector<int>& nums, int i) {
        if (i == nums.size()) {
            result.push_back(nums);
            return;
        }

        for (int j = i; j < nums.size(); j++) {
            if (j > i && nums[j] == nums[i])
                continue;
            swap(nums[i], nums[j]);
            backtrack(nums, i + 1);
        }

        for (int j = nums.size() - 1; j > i; j--) {
            swap(nums[i], nums[j]);
        }
    }

    vector<vector<int>> permuteUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        backtrack(nums, 0);
        return result;
    }
};